examples/model_selection/plot_grid_search_digits.py

```diff
@@ -13,56 +13,177 @@
 
 More details on tools available for model selection can be found in the
 sections on :ref:`cross_validation` and :ref:`grid_search`.
-
 """
 
+# %%
+# The dataset
+# -----------
+#
+# We will work with the `digits` dataset. The goal is to classify handwritten
+# digits images.
 from sklearn import datasets
-from sklearn.model_selection import train_test_split
-from sklearn.model_selection import GridSearchCV
-from sklearn.metrics import classification_report
-from sklearn.svm import SVC
 
-# Loading the Digits dataset
-X, y = datasets.load_digits(return_X_y=True)
+digits = datasets.load_digits()
+
+# %%
+# To apply a classifier on this data, we need to flatten the images.
+# Each image of 8 by 8 pixels needs to be transformed to a vector of 64 pixels.
+# Thus, we will get a final data array of shape (n_images, n_pixels)
+n_samples = len(digits.images)
+X = digits.images.reshape((n_samples, -1))
+y = digits.target == 7
```

Comment:
This instruction sets the problem to a binary classification one.

I think it would be nice to mention that at the top, or to use a multi-class classification problem.